libs/core/tests/sharing_tests.rs

```diff
@@ -742,6 +949,43 @@ fn delete_link_to_share() {
     assert::all_pending_shares(&cores[1], &["folder"]);
 }
 
+#[test]
+fn delete_link_target_to_share() {
+    let cores = vec![test_core_with_account(), test_core_with_account(), test_core_with_account()];
+    let accounts = cores
+        .iter()
+        .map(|core| core.get_account().unwrap())
+        .collect::<Vec<_>>();
+    let roots = cores
+        .iter()
+        .map(|core| core.get_root().unwrap())
+        .collect::<Vec<_>>();
+
+    let folder = cores[0]
+        .create_file("folder", roots[0].id, FileType::Folder)
+        .unwrap();
+    cores[0]
+        .share_file(folder.id, &accounts[1].username, ShareMode::Write)
+        .unwrap();
+
+    cores[0].sync(None).unwrap();
+    cores[1].sync(None).unwrap();
+
+    assert::all_pending_shares(&cores[1], &["folder"]);
+
+    cores[1].create_link_at_path("link/", folder.id).unwrap();
+
+    assert::all_pending_shares(&cores[1], &[]);
+
+    cores[1].delete_file(folder.id).unwrap();
+
+    cores[1].sync(None).unwrap();
+    cores[0].sync(None).unwrap();
+
+    assert::all_paths(&cores[0], &["/", "/folder/"]);
+
+    assert::all_pending_shares(&cores[1], &["folder"]);
```

Comment:
an FYI, deleting the link puts it back into a pending status for you, as this test demonstrates. 

deleting the actual file unshares it from you, it should also get rid of the link. Travis thinks there is a test for this behavior.

Reply:
okay i deleted that test

Reply:
you should confirm that there is a test for both sides of this, deleting the link as well as deleting the share

Reply:
deleting a share deletes a link (demonstrated by test `delete_accepted_share`) but deleting a file does not and need not unshare it, since non-deletion updates to deleted files are thrown away anyway

Reply:
I'll write some good docs on sync and force the team to read and discuss them when the editor is in a better state 😉